### PR TITLE
OLH-1621 Call the MFA methods API when a user visits the security page

### PR DESCRIPTION
**Pull request by aenciso**
## Proposed changes
[OLH-1621] Call the MFA methods API when a user visits the security page
### What changed

Added middleware to call MFA methods and store the result in the session

### Why did it change

To enable the FE to access available MFA methods

### Related links

https://github.com/govuk-one-login/account-management-stubs/pull/36

### Environment variables or secrets
- [ ] METHOD_MANAGEMENT_BASE_URL

- [ ] Application configuration is up-to-date
- [ ] Documented in the README
- [ ] Added to deployment steps
- [ ] Added to local startup config

## Testing

Unit and integration testing. Once stub is deployed, it needs to be tested against stub as well.

## How to review

[OLH-1621]: https://govukverify.atlassian.net/browse/OLH-1621?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .env.sample [hunk 1]
```diff
@@ -37,3 +37,4 @@ ENVIRONMENT=
 VERIFY_ACCESS_VALUE=
 # subjectId returned from oidc stub
 MY_ONE_LOGIN_USER_ID=user_id
+MFA_ENDPOINT_URL=https://method-management-v1-stub.home.dev.account.gov.uk
```
Comment (andrew-moores) [inline .env.sample:40]:
```suggestion
METHOD_MANAGEMENT_URL=https://method-management-v1-stub.home.dev.account.gov.uk
```

File: docker-compose.yml [hunk 2]
```diff
@@ -64,6 +64,7 @@ services:
       - ACCOUNT_ID=${ACCOUNT_ID}
       - ENVIRONMENT=${ENVIRONMENT}
       - VERIFY_ACCESS_VALUE=${VERIFY_ACCESS_VALUE}
+      - MFA_ENDPOINT_URL=${MFA_ENDPOINT_URL}
```
Comment (andrew-moores) [inline docker-compose.yml:67]:
```suggestion
      - METHOD_MANAGEMENT_URL=${METHOD_MANAGEMENT_URL}
```

File: src/config.ts [hunk 3]
```diff
@@ -238,3 +238,7 @@ export function showContactEmergencyMessage(): boolean {
 export function getContactEmailServiceUrl(): string {
   return process.env.CONTACT_EMAIL_SERVICE_URL;
 }
+
+export function getMfaServiceUrl(): string {
+  return process.env.MFA_ENDPOINT_URL;
```
Comment (andrew-moores) [inline src/config.ts:243]:
```suggestion
  return process.env.METHOD_MANAGEMENT_URL;
```

File: .env.sample (hunk 1 above)
Comment (aenciso) [reply .env.sample]: Done

File: src/config.ts (hunk 3 above)
Comment (aenciso) [reply src/config.ts]: Done

File: src/utils/types.ts [hunk 4]
```diff
@@ -124,3 +124,39 @@ export const allowedTxmaEvents: Array<string> = [
   "AUTH_AUTH_CODE_ISSUED",
   "AUTH_IPV_AUTHORISATION_REQUESTED",
 ];
+
```
[the remark this answers is not in the document]
Comment (aenciso) [reply src/utils/types.ts]: Done

File: docker-compose.yml (hunk 2 above)
Comment (aenciso) [reply docker-compose.yml]: Done

**Review by andrew-moores — APPROVED** [review]

**Comment by andrew-moores** [comment]
Do we need a feature flag so we can disable calling the new method management api in prod until its available?

**Review by andrew-moores — APPROVED** [review]

**Pull request merged**